app/src/main/java/org/mozilla/fenix/gleanplumb/state/MessagingMiddleware.kt

```diff
@@ -144,6 +144,9 @@ class MessagingMiddleware(
         if (actualMessageToShow?.id == oldMessage.id) {
             context.dispatch(UpdateMessageToShow(updatedMessage))
         }
-        return removeMessage(context, oldMessage) + updatedMessage
+        val oldMessageIndex = context.state.messaging.messages.indexOfFirst { it.id == updatedMessage.id }
```

Comment:
```suggestion
        val oldMessageIndex = context.state.messaging.messages.indexOfFirst { it.id == oldMessage.id }
```

The ID is the same regardless of the two messages, but since the binding is `oldMessageIndex`, did you mean to use `oldMessage`?

Reply:
No, actually we want to check if the id of the `updatedMessage` matches to the id of any item of the `messages` list. This item is actually the `oldMessage` which will always be present in the list.